lib/cylc/task_proxy.py

```diff
@@ -316,9 +320,6 @@ def __init__(
             self.TABLE_TASK_STATES: [],
         }
 
-        # TODO - should take suite name from config!
-        self.suite_name = os.environ['CYLC_SUITE_NAME']
```

Comment:
Doesn't need to be an instance variable, and this was an insane way of getting the suite name.